toolkit/scripts/sodiff/mariner-sodiff.sh

```diff
@@ -9,41 +9,81 @@ repo_file_path="$2"
 mariner_version="$3"
 sodiff_out_dir="$4"
 sodiff_log_file="${sodiff_out_dir}/sodiff.log"
+current_os=$(cat /etc/os-release | grep ^ID | cut -d'=' -f2)
 
 # Setup output dir
 mkdir -p "$sodiff_out_dir"
+# Empty the log file
+echo > "$sodiff_log_file"
 
-# Prepare mariner/ubuntu compatibility calls
+function makecache_with_common {
+    if [[ "$current_os" == mariner ]]; then
+        # Mariner uses DNF repoquery command
+        DNF_COMMAND=dnf
+        # Cache RPM metadata
+        >/dev/null dnf -c "$1" --releasever $mariner_version -y makecache
+    else
+        # Ubuntu uses repoquery command from yum-utils
+        DNF_COMMAND=
+        # Cache RPM metadata
+        # Ubuntu does not come with gpgcheck plugin for yum
+        >/dev/null yum -c "$1" --releasever $mariner_version -y --nogpgcheck makecache
+    fi
+}
+
+function set_common_options {
+    if [[ "$current_os" == mariner ]]; then
+        # Mariner uses DNF repoquery command
+        DNF_COMMAND=dnf
+    else
+        # Ubuntu uses repoquery command from yum-utils
+        DNF_COMMAND=
+    fi
+    common_options="-c $repo_file_path --releasever $mariner_version"
```

Comment:
If a variable is used outside the scope of this function, let's use upper case letters to show it's not local.